widgets/indicator/options.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2019 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package indicator
+
+// options.go contains configurable options for Speedometer.
+
+import (
+	"github.com/mum4k/termdash/align"
+	"github.com/mum4k/termdash/cell"
+)
+
+// Option is used to provide options.
+type Option interface {
+	// set sets the provided option.
+	set(*options)
+}
+
+// option implements Option.
+type option func(*options)
+
+// set implements Option.set.
+func (o option) set(opts *options) {
+	o(opts)
+}
+
+// options holds the provided options.
+type options struct {
+	status       bool
+	textCellOpts []cell.Option
+	cellOpts     []cell.Option
+
+	labelCellOpts []cell.Option
+	labelAlign    align.Horizontal
+	label         string
+}
+
+// validate validates the provided options.
+func (o *options) validate() error {
+	return nil
+}
+
+// newOptions returns options with the default values set.
+func newOptions() *options {
+	return &options{
+		status: true,
+		textCellOpts: []cell.Option{
+			cell.FgColor(cell.ColorRed),
+			cell.BgColor(cell.ColorDefault),
+		},
+		labelAlign: DefaultLabelAlign,
+	}
+}
+
+// TextCellOpts sets cell options on cells that contain the displayed text
+// progress.
+func TextCellOpts(cOpts ...cell.Option) Option {
```

Comment:
Does it make sense to pass in cell options here? They allow to pass both foreground and background colors, but the indicator isn't always a text character anymore.

Should we rename this to just Color? I.e. an option that sets the color of the indicator?

Looking below we also have CellOpts. It isn't very clear which one is for what and the text on them still mentions the Donut widget (probably and copy&paste).

Can we try to cleanup this options file before we move on with the review since this is also part of the public API.

Reply:
Good point, I've spit the options up into Color and LabelColor for readability.